library/Hasql/Pool.hs

```diff
@@ -85,8 +85,9 @@ release Pool {..} =
     newReuse <- newTVar True
     writeTVar poolReuse newReuse
     conns <- flushTQueue poolConnectionQueue
-    modifyTVar' poolCapacity (+ (length conns))
-    return $ forM_ conns Connection.release
+    return $ forM_ conns $ \conn -> do
+      Connection.release conn
+      atomically $ modifyTVar' poolCapacity succ
```

Comment:
alternatively,

```
    return $ do
      forM_ conns Connection.release
      atomically $ modifyTVar' poolCapacity (+ (length conns))
```

but I thought it better to make extra capacity available as quickly as possible, at the expense of higher locking overhead